app/models/comment.rb

```diff
@@ -344,7 +317,8 @@ def update_score_and_recalculate!(score_delta, flag_delta)
       UPDATE comments SET
         score = (select coalesce(sum(vote), 0) from votes where comment_id = comments.id),
         flags = (select count(*) from votes where comment_id = comments.id and vote = -1),
-        confidence = #{self.calculated_confidence}
+        confidence = #{self.calculated_confidence},
+        confidence_order = concat(lpad(char(65536 - floor(((confidence - -0.2) * 65535) / 1.2) using binary), 2, '0'), char(id & 0xff using binary))
```

Comment:
Is there a way to keep the math in one place to calculate confidence order? Maybe this is a good place to use a UDF to hold the confidence to confidence order calculation and use it here and in `#assign_initial_confidence`?

Reply:
Rails does not have good support for UDF/stored procedures, so I worry there'd be pain to maintain that integration.